find-modules/rules_engine.py

```diff
@@ -80,7 +84,10 @@ def __init__(self, db, rule_number, fn, pattern_zip):
         try:
             groups = ["(?P<{}>{})".format(name, pattern) for pattern, name in pattern_zip]
             groups = _SEPARATOR.join(groups)
-            self.matcher = re.compile(groups)
+            #
+            # We'll use re.match to anchor the start of the match, and so need a $ to anchor the end.
```

Comment:
I'm not so familiar with the python regex system. Could you give an example of what this changes?